dotnet/src/webdriver/BiDi/BrowsingContext/SetViewportCommand.cs

```diff
@@ -18,13 +18,14 @@
 // </copyright>
 
 using OpenQA.Selenium.BiDi.Communication;
+using System.Text.Json.Serialization;
 
 namespace OpenQA.Selenium.BiDi.BrowsingContext;
 
 internal sealed class SetViewportCommand(SetViewportParameters @params)
     : Command<SetViewportParameters, SetViewportResult>(@params, "browsingContext.setViewport");
 
-internal sealed record SetViewportParameters(BrowsingContext Context, Viewport? Viewport, double? DevicePixelRatio) : Parameters;
+internal sealed record SetViewportParameters(BrowsingContext Context, [property: JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)] Viewport? Viewport, double? DevicePixelRatio) : Parameters;
```

Comment:
Nice catch, we already started doing it in another places. Let's find all occurences of this type (`property: type / null`) and mark all with this attribute.

Reply:
Yes, it always sends `null`:
```
19:46:18.576 TRACE WebSocketTransport: BiDi SND --> {"id":4,"method":"browsingContext.setViewport","params":{"context":"EAEFDFDBD3A2386DD196A128BE21FF1E","viewport":null}}
```

But I am more interesting whether it's possibly to send `null` only and only when user explicitly set it to `null`. Is it poossible?

Reply:
Filed this interesting case to spec: https://github.com/w3c/webdriver-bidi/issues/1039

Reply:
Resolving this thread, let's start new discussions.